SIMSBIOHUB-353/325: Survey form updates

al-rosenthal opened the pull request:
## Links to Jira Tickets

[SIMSBIOHUB-325](https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-325)
[SIMSBIOHUB-353](https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-353)

## Description of Changes

- Site selection strategies is now required for surveys 
- Removed field method from surveys

## Testing Notes

- You should be able to create and edit new and existing surveys without issue

al-rosenthal committed:
added required tag to site selection form and open api spec

al-rosenthal committed:
removed references to field method from survey

al-rosenthal committed:
fixing code smells

curtisupshall commented on api/src/paths/project/{projectId}/survey/create.ts:
```diff
@@ -202,7 +202,8 @@ POST.apiDoc = {
                   type: 'string'
                 },
                 field_method_id: {
```
It's probably in our best interest to remove and field mthod code entirely, and not just allow it to be null. Thoughts?

JeremyQuartech replied:
Yeah I agree, I think it could cause some confusing leaving this in here.

al-rosenthal replied:
good call, removed field method from the open api specs

curtisupshall replied:
> good call, removed field method from the open api specs

What I mean is, we should probably remove any and all mention of `field_method` from the codebase. Example:

![image](https://github.com/bcgov/biohubbc/assets/18599267/e02d96c8-aa10-4ebf-926c-3a182a6381df)

al-rosenthal replied:
I have since removed field method from the project 😅

JeremyQuartech reviewed (commented):
It might make sense to update the db seed file, adding a value for site selection strategy, seeing as it should be required, and removing the value for field method id. A deprecation comment in the db column for field_method_id could be useful too.

al-rosenthal committed:
removing field method from open api, depreciating table

al-rosenthal committed:
added depreciation tags to table and column

al-rosenthal committed:
clean up

al-rosenthal committed:
removed remaining field methods from open api specs

al-rosenthal committed:
fixed broken test

al-rosenthal committed:
removed field method and added site strategy to project seed

al-rosenthal commented:
> It might make sense to update the db seed file, adding a value for site selection strategy, seeing as it should be required, and removing the value for field method id. A deprecation comment in the db column for field_method_id could be useful too.

Good idea. I've removed the field method from the seed file and added the site strategy to the survey seed process

JeremyQuartech reviewed (approved):
Tested locally, everything looks good to me 👍

al-rosenthal committed:
removed remaining mentions of field method

al-rosenthal committed:
fixed more tests

al-rosenthal committed:
Merge branch 'dev-staging' into SIMSBIOHUB-353: changes so nice I made them twice

al-rosenthal committed:
ran fix

curtisupshall reviewed (approved):
Code looks good to me. Tested this locally and everything works as expected. 👍
